### PR TITLE
Reconfigure workspaces, build on Travis

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

**Comment by thomcc** [comment]
We should merge this because without it the build is broken due to having multiple workspaces. (This is my fault, sorry...)

**Comment by eoger** [comment]
What do we want to do with this? Is still still needed?

**Comment by vladikoff** [comment]
@eoger what do we need to do to make it build in this log https://travis-ci.org/mozilla/application-services/jobs/379026612 ?

**Comment by eoger** [comment]
Looks like I've managed to fix the build.  
I've also cached `libs/ios/` since it's very slow to build, but I think in the future we should probably have a build script that only targets x86_64 (for local/travis-ci testing).

One more thing, @vladikoff is there a particular reason we do `cargo build --release` and  `cargo test` instead of just doing the latter? It makes our build even slower because for some reason it re-compiles every cargo dependency.

**Comment by vladikoff** [comment]
> I've also cached libs/ios/ since it's very slow to build

👍

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+language: rust
+os: osx
```
Comment (vladikoff) [inline .travis.yml]: Interesting!

**Comment by vladikoff** [comment]
> One more thing, @vladikoff is there a particular reason we do cargo build --release and cargo test instead of just doing the latter?

Fixed it up

File: .travis.yml (hunk 1 above)
Comment (eoger) [reply .travis.yml]: Yeah we need that for lipo, but like in said in https://github.com/mozilla/application-services/pull/6#issuecomment-389378704 we don't really need these lipoed libraries.

**Comment by vladikoff** [comment]
oof

image

**Pull request merged**